NuRadioMC/SignalProp/analyticraytracing.py

```diff
@@ -153,215 +431,61 @@ def __init__(self, medium, attenuation_model="SP1",
         if overwrite_speedup is not None:
             self._use_optimized_calculation = overwrite_speedup
         self.use_cpp = use_cpp
+        if compile_numba:
+            if numba_available:
+                global get_reflection_point,obj_delta_y_square,get_delta_y
+                global get_y_turn, get_y_with_z_mirror,get_turning_point
+                global get_gamma, get_y, get_C0_from_log
+                global  get_z_unmirrored, get_y_diff, n
+                try :
+                
+                    get_reflection_point = jit(get_reflection_point, nopython=True, cache=True)
+                    obj_delta_y_square = jit(obj_delta_y_square, nopython=True, cache=True)
+                    get_delta_y = jit(get_delta_y, nopython=True, cache=True)
+                    get_y_turn = jit(get_y_turn, nopython=True, cache=True)
+                    get_y_with_z_mirror = jit(get_y_with_z_mirror, nopython=True, cache=True)
+                    get_turning_point = jit(get_turning_point, nopython=True, cache=True)
+                    get_gamma = jit(get_gamma, nopython=True, cache=True)
+                    get_y = jit(get_y, nopython=True, cache=True)
+                    get_C0_from_log = jit(get_C0_from_log, nopython=True, cache=True)
+                    get_z_unmirrored = jit(get_z_unmirrored, nopython=True, cache=True)
+                    n = jit(n, nopython=True, cache=True)
+                    self.use_cpp = False
+                except:
```

Comment:
Good point, the kind of error that we could expect here are compilation errors that could be caused by an outdated version of Numba, or when the method has been modified without following Numba guidelines.
In any case if this fails, the faulty method and consequently the whole class might not be functional - it might be better to altogether remove the try/except clause and let whatever error happen.

Reply:
I remember now, for some reason at initialisation the class tries to run this section twice and generates an error, restoring it.